index.js

```diff
@@ -31,6 +34,50 @@ app.post("/message", (req, res) => {
   res.json(message);
 });
 
+app.post("/message/:id/reaction", (req, res) => {
+  if (!req.body.user) {
+    return res.status(400).json({ error: "Missing user" });
+  }
+  if (!req.body.reaction) {
+    return res.status(400).json({ error: "Missing reaction" });
+  }
+  // handling duplicate user reactions
+  function compareReactions(e) {
+    const userReaction = req.body.reaction;
+    const user = req.body.user;
+    const messageId = parseInt(req.params.id);
+    if (
+      e.user === user &&
+      e.reaction === userReaction &&
+      e.messageId === messageId
+    ) {
+      return true;
+    }
+  }
+  if (reactions.some(compareReactions)) {
+    res.status(400).json({ error: " Duplicate reaction" });
+  } else {
+    const reaction = {
+      id: reactionNextId++,
+      messageId: parseInt(req.params.id),
+      user: req.body.user,
+      reaction: req.body.reaction,
+      createdAt: new Date(),
+    };
+    reactions.push(reaction);
+    res.json(reaction);
+  }
+});
+
+app.get("/message/:id/reactions", (req, res) => {
```

Comment:
Could we just include this in the `/message/:id` endpoint so that the client doesn't need to make a request for each new message it gets to find the emojis?

Reply:
For example, make the messages response in `/messages/` return something like this:

```
{
    id: 1,
    user: 'aaron',
    message: 'hi codefriends',
    createdAt: '...',
    reactions: [
        {
            id: 1,
            reaction: '😊',
            user: 'brandon',
            createdAt: '...'
        }
    ]
}
```

Reply:
ok, will do